src/simulation/m_time_steppers.fpp

```diff
@@ -980,6 +1004,187 @@ contains
         end if
 
     end subroutine s_time_step_cycling
+
+    !> (Adaptive) 4th/5th order Runge—Kutta–Cash–Karp (RKCK) time-stepping algorithm (Cash J. and Karp A., 1990)
+        !!      Method for initial value problems with rapidly varying RHS. A maximum error between the 4th and 5th
+        !!      order Runge-Kutta-Cash-Karp solutions for the same time step size is calculated. If the error is
+        !!      smaller than a tolerance, then the algorithm employs the 5th order solution, while if not, both
+        !!      eulerian/lagrangian variables are re-calculated with a smaller time step size.
+        !! @param t_step Current time-step
+        !! @param hdid Advanced time increment (adaptive time stepping)
+    subroutine s_4th_5th_order_rkck(t_step, hdid, time_prev, time_avg)
+
+        integer, intent(in) :: t_step
+        real(kind(0d0)), intent(inout) :: hdid
+        real(kind(0d0)), intent(in) :: time_prev
+        real(kind(0d0)), intent(inout) :: time_avg
+
+        real(kind(0.d0)) :: newtime, errmax_glb, dttarget, qtime
+        real(kind(0.d0)) :: RKh, RKh_glb, htemp, SAFETY = 0.9d0, PGROW = -0.2d0, &
```

Comment:
safety, pgrow, pshrnk, errcon should go in `m_constants`. sorry I didn't catch this earlier.

Reply:
fixed